src/event/metric.rs

```diff
@@ -18,34 +18,49 @@ pub struct Metric {
 
 #[derive(Debug, Hash, Clone, PartialEq, Deserialize, Serialize, is_enum_variant)]
 #[serde(rename_all = "snake_case")]
+/// A metric may be an incremental value, updating the previous value of
+/// the metric, or absolute, which sets the reference for future
+/// increments.
 pub enum MetricKind {
     Incremental,
     Absolute,
 }
 
 #[derive(Debug, Clone, PartialEq, Deserialize, Serialize, is_enum_variant)]
 #[serde(rename_all = "snake_case")]
+/// A MetricValue is the container for the actual value of a metric.
 pub enum MetricValue {
-    Counter {
-        value: f64,
-    },
-    Gauge {
-        value: f64,
-    },
-    Set {
-        values: BTreeSet<String>,
-    },
+    /// A Counter is a simple value that can not decrease except to
+    /// reset it to zero.
+    Counter { value: f64 },
+    /// A Gauge represents a sampled numerical value.
+    Gauge { value: f64 },
+    /// A Set contains a set of (unordered) unique values for a key.
+    Set { values: BTreeSet<String> },
+    /// A Distribution contains a set of sampled values paired with the
+    /// rate at which they were observed.
     Distribution {
         values: Vec<f64>,
         sample_rates: Vec<u32>,
         statistic: StatisticKind,
```

Comment:
What does this do? Does it make sense to keep this value around?

Reply:
The `statistic` field comes from the statsd source, and is used by many of the metrics sinks. I don't think we can just drop it.